### PR TITLE
Add tls-everywhere

### PR DESCRIPTION
**Pull request by asyedham**

### CODE REVIEW

**Comment by masco** [comment]
Can one of the admins verify this patch?

File: overcloud.yml [hunk 1]
```diff
@@ -133,12 +133,30 @@
...
+
 
       - name: run tripleo-overcloud deploy with extra templates
         shell: |
             source .venv/bin/activate
             infrared tripleo-overcloud -vvv --version {{ osp_release }} --deployment-timeout 240 --build {{ osp_puddle }}  --deployment-files {{ nic_configs }} --introspect no --tagging no --deploy yes --controller-nodes {{ controller_count }} --compute-nodes {{ compute_count }} --overcloud-templates {{ infrared_dir }}/plugins/tripleo-overcloud/vars/overcloud/templates/extra.yml --network-protocol ipv4 --network-backend {{ network_backend }} {{ network_type }} true --public-network false > {{ log_directory }}/overcloud_deploy.log 2>&1
         args:
             chdir: "{{ infrared_dir }}"
-        when: ( extra_templates is defined and extra_templates|length>0 ) or ( parameter_defaults is defined and parameter_defaults|length>0 )
+        when: tls_everywhere !=true and ( extra_templates is defined and extra_templates|length>0 ) or ( parameter_defaults is defined and parameter_defaults|length>0 )
```
Comment (smalleni) [inline overcloud.yml]: Can also use the "not" notation like not tls_everywhere

File: freeipa/group_vars/freeipa.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+freeipa_overcloud_deploy_script: "~/overcloud_deploy.sh"
+freeipa_overcloud_stack: "{{ (install|default({})).get('overcloud', {}).get('stack', 'overcloud') }}"
+freeipa_source_dir: "{{ (install|default({})).get('deployment', {}).get('files', 'virt') }}"
+freeipa_templates_basedir: "{{ hostvars[groups['undercloud'][0]].ansible_user_dir }}/{{ freeipa_source_dir | basename }}"
+freeipa_heat_templates_basedir: "{{ (install|default({})).get('heat', {}).get('templates', {}).get('basedir', '/usr/share/openstack-tripleo-heat-templates') }}"
+freeipa_use_ceph: "{{ ( True if (install.storage|default({})).backend|default('') == 'ceph' else False ) }}"
+freeipa_network_environment_file: "{{ freeipa_templates_basedir }}/network/network-environment.yaml"
+#freeipa_network_environment_file:
+freeipa_network_environment_v6_file: "{{ freeipa_templates_basedir }}/network/network-environment-v6.yaml"
+freeipa_forwarder: 10.1.32.3
```
Comment (smalleni) [inline freeipa/group_vars/freeipa.yml:40]: Hardcoded

File: group_vars/all.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@ infrared_workspaces_dir: "{{ ansible_user_dir }}/.infrared/.workspaces"
 infrared_hosts_file: "{{ infrared_workspaces_dir }}/active/hosts"
 instackenv_file: "{{ playbook_dir }}/instackenv.json"
 undercloud_conf: "{{ ansible_user_dir }}/undercloud.conf"
-log_directory: "{{ playbook_dir }}/jetpack/logs"
+log_directory: ""{{ playbook_dir }}/logs"
```
Comment (smalleni) [inline group_vars/all.yml]: Don't need to change this

File: add_undercloud_to_inventory.yml [hunk 4]
```diff
@@ -16,6 +16,24 @@
...
+        name: "localhost"
+        groups: "local"
+        ansible_user: "root"
+        ansible_python_interpreter: "{{ python_interpreter }}"
+
+    - name: add freeipa_host to inventory file
+      add_host:
+        name: "freeipa"
+        groups: "freeipa"
+        ansible_host: "{{ freeipa_hostname }}"
```
Comment (smalleni) [inline add_undercloud_to_inventory.yml:31]: How is " freeipa.redhat.local" which is freeipa_hostname resolved?

**Review by smalleni — CHANGES_REQUESTED** [review]

File: add_undercloud_to_inventory.yml (hunk 4 above)
Comment (asyedham) [reply add_undercloud_to_inventory.yml:31]: The hostname is changed to freeipa.redhat.local during the undercloud deployment and is copied into the /etc/hosts to resolve it

File: group_vars/all.yml (hunk 3 above)
Comment (asyedham) [reply group_vars/all.yml]: I changed it from "{{ playbook_dir }}/ljetpack/logs"  to "{{ playbook_dir }}/logs" as it unnecessarily creates jetpack dir and then inside creates the log dir

File: freeipa/group_vars/freeipa.yml (hunk 2 above)
Comment (asyedham) [reply freeipa/group_vars/freeipa.yml:40]: yeah because these are group_vars specific to freeipa host

File: group_vars/all.yml (hunk 3 above)
Comment (smalleni) [reply group_vars/all.yml]: Ha, I see. Good catch.

File: overcloud.yml (hunk 1 above)
Comment (asyedham) [reply overcloud.yml]: So when we set tls_everywhere to true in group_vars/all.yml. Then not tls_everywhere will result it into false

File: freeipa/group_vars/freeipa.yml (hunk 2 above)
Comment (smalleni) [reply freeipa/group_vars/freeipa.yml:40]: What is the freeipa host, is it part of the allcoation? 
10.1.32.3 is the lab dns isnt it?

Comment (asyedham) [reply freeipa/group_vars/freeipa.yml:40]: > What is the freeipa host, is it part of the allcoation?
Yeah. The first node in instackenv.json is undercloud and next is the freeipa. It is configured with a IP 192.168.24.251 on 2nd interface
> 10.1.32.3 is the lab dns isnt it?
yeah.

**Comment by masco** [comment]
@asyedham could you rebase this patch. i got machines and I am going to test

**Comment by asyedham** [comment]
@masco Have resolved the merge conflicts. Please have a look

File: overcloud.yml [hunk 5]
```diff
@@ -133,12 +133,30 @@
             infrared tripleo-overcloud -vvv --version {{ osp_release }} --deployment-timeout 240 --deployment-files {{ nic_configs }} --introspect no --tagging no --deploy yes --controller-nodes {{ controller_count }} --compute-nodes {{ compute_count }} --network-protocol ipv4 --network-backend {{ network_backend }} {{ network_type }} true --public-network false > {{ log_directory }}/overcloud_deploy.log 2>&1
         args:
             chdir: "{{ infrared_dir }}"
-        when: extra_templates is not defined and parameter_defaults is not defined
+        when: tls_everywhere != true and extra_templates is not defined and parameter_defaults is not defined
+
+      - name: run tripleo-overcloud deploy tls
+        shell: |
+            source .venv/bin/activate
+            infrared tripleo-overcloud -vvv --version {{ osp_release }} --deployment-files {{ nic_configs }} --introspect no --tagging no --deploy yes --controller-nodes {{ controller_count }} --compute-nodes {{ compute_count }} --tls-everywhere true > {{ log_directory }}/overcloud_deploy.log 2>&1
```
Comment (masco) [inline overcloud.yml]: In the above command at line no. 133, if we pass tls-everywhere true will it work or we have to cut down those extra parameters?

File: undercloud.yml [hunk 6]
```diff
@@ -37,10 +45,45 @@
...
+          path: "{{ infrared_tls_dir }}/tasks/prepare_undercloud.yml"
+          insertbefore:  Prepare novajoin to work
+          block: |
+            ##ipa-client
+              - name: execute ipa-client
+                shell: |
+                    sudo ipa-client-install --principal=admin --password={{ freeipa_admin_password }} --server={{ freeipa_server_hostname }} --domain={{ freeipa_cloud_domain }} --realm={{ freeipa_realm }} --hostname={{ freeipa_undercloud_hostname }} --unattended --force-join
+                changed_when: false
+                ignore_errors: true
+        when: tls_everywhere == true
```
Comment (masco) [inline undercloud.yml]: NIT: this all tasks can be put in a single block.

File: undercloud.yml [hunk 7]
```diff
@@ -52,3 +95,17 @@
...
+          source {{ infrared_dir }}/.venv/bin/activate
+          infrared tripleo-undercloud -vv \
+          --version {{ osp_release }} \
+          --build {{ osp_puddle }} \
+          --images-task rpm \
+          --config-file {{ undercloud_conf }} {{ container_params }} --tls-everywhere true > {{ log_directory }}/undercloud_install.log 2>&1
+        args:
+            chdir: "{{ infrared_dir }}"
+        changed_when: false
+        when: tls_everywhere == true
```
Comment (masco) [inline undercloud.yml]: AFIK --tls-everywhere {{ tls_everywhere}} in the previous task will work

Comment (asyedham) [reply undercloud.yml]: Done. Could you please review and test.

**Pull request closed without merging**